ciao-cli/configuration.go

```diff
@@ -0,0 +1,237 @@
+	}
+
+	url := buildCiaoURL("configuration")
+	req = types.ConfigRequest{
+		Element: string(cmd.element),
+		Value:   cmd.value,
+	}
+	b, err := json.Marshal(req)
+	if err != nil {
+		fatalf(err.Error())
+	}
+	body := bytes.NewReader(b)
+	ver := api.ClusterConfigV1
+
+	resp, err := sendCiaoRequest("POST", url, nil, body, &ver)
+	if err != nil {
+		return err
+	}
+
+	err = unmarshalHTTPResponse(resp, &response)
+	if err != nil {
+		return err
+	}
+	fmt.Println(response.Response)
+	return nil
+}
+
+func validConfigValue(s string, t string) bool {
+	switch t {
+	case "scheduler.storage_uri":
+		return validConfigURI(s, "file")
+	case "identity_service.url":
+		return validConfigURI(s, "https")
+	case "controller.compute_port":
+		return validConfigNumber(s)
+	case "launcher.disk_limit", "launcher.mem_limit":
+		return validConfigBoolean(s)
+	case "identity_service.type":
+		return s == "keystone"
+	}
+	return false
+}
+
+// validConfigNumber checks that current input is a sane integer number
+func validConfigNumber(s string) bool {
+	if _, err := strconv.Atoi(s); err == nil {
+		return true
+	}
+	return false
+}
+
+func sanatizeBoolean(s string) string {
+	s = strings.ToLower(s)
+	switch s {
+	case "1", "t", "true":
+		return "true"
+	case "0", "f", "false":
+		return "false"
+	}
+	return ""
+}
+
+// validConfigBoolean returns true if the input (s) is correct value
+// for a boolean and its representation for the configuration yaml payload
+func validConfigBoolean(s string) bool {
+	s = strings.ToLower(s)
+	switch s {
+	case "1", "t", "true", "0", "f", "false":
+		return true
+	}
+	return false
+}
+
+// validConfigURI check correctness of the URI given, evaluating
+// if the string to be analized matches with the expected scheme
+// and meets the URI elements needed for scheme
+func validConfigURI(s string, scheme string) bool {
+	uri, err := url.Parse(s)
+	if err != nil {
+		return false
+	}
+	if scheme != uri.Scheme {
+		return false
+	}
+	switch scheme {
+	case "file":
+		return uri.Path != ""
+	case "https":
+		// check hostname is explicit (e.g: "https://:35357" is invalid)
+		return (uri.Host != "") && (strings.HasPrefix(uri.Host, ":") == false)
+	}
+	return false
+}
+
+type configShowCommand struct {
+	Flag flag.FlagSet
+}
+
+func (cmd *configShowCommand) usage(...string) {
+	fmt.Fprintf(os.Stderr, `
+Usage:
+
+ciao-cli config show
+
+Show current cluster configuration
+`)
+	os.Exit(2)
+}
+
+func (cmd *configShowCommand) parseArgs(args []string) []string {
+	cmd.Flag.Usage = func() { cmd.usage() }
+	cmd.Flag.Parse(args)
+	return cmd.Flag.Args()
+}
+
+func (cmd *configShowCommand) run(args []string) error {
+	var response types.ConfigShowResponse
+
+	ver := api.ClusterConfigV1
+	url := buildCiaoURL("configuration")
+	resp, err := sendCiaoRequest("GET", url, nil, nil, &ver)
+	if err != nil {
+		return err
+	}
+
+	err = unmarshalHTTPResponse(resp, &response)
+	if err != nil {
+		return err
+	}
+	fmt.Print(response.Configuration)
```

Comment:
We should add template support here, i.e., ciao-cli config show -f.  See the other show commands.